great_expectations/validator/metrics_calculator.py

```diff
@@ -1,18 +1,23 @@
+from __future__ import annotations
+
 import logging
-from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, Tuple, Type, Union
 
 import great_expectations.exceptions as gx_exceptions
-from great_expectations.execution_engine import (
-    ExecutionEngine,
-    PandasExecutionEngine,
-    SparkDFExecutionEngine,
-    SqlAlchemyExecutionEngine,
-)
+from great_expectations.util import load_class
 from great_expectations.validator.computed_metric import MetricValue  # noqa: TCH001
 from great_expectations.validator.exception_info import ExceptionInfo  # noqa: TCH001
 from great_expectations.validator.metric_configuration import MetricConfiguration
 from great_expectations.validator.validation_graph import ValidationGraph
 
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import (
+        ExecutionEngine,
+        PandasExecutionEngine,
```

Comment:
these engine-specific imports should be removed

Reply:
@donaldheppner This is formally outside of the scope and can be done as a separate tight pull request.